src/lib.rs

```diff
@@ -0,0 +1,186 @@
+}
+
+// Struct used to communicate a tracing configuration to C. Must stay in sync with the C code.
+#[repr(C)]
+struct TracerConf {
+    target_pid: pid_t,
+    trace_filename: *const c_char,
+    map_filename: *const c_char,
+    data_bufsize: size_t,
+    aux_bufsize: size_t,
+}
+
+/// Represents an instance of the Intel PT tracer.
+#[derive(Default)]
+pub struct Tracer {
+    /// Filename to store the trace to.
+    trace_filename: CString,
+    /// PID to trace.
+    target_pid: pid_t,
+    /// Data buffer size, in pages. Must be a power of 2.
+    data_bufsize: size_t,
+    /// Aux buffer size, in pages. Must be a power of 2.
+    aux_bufsize: size_t,
+    /// Opaque C pointer representing the tracer context.
+    tracer_ctx: Option<*const c_void>,
+}
+
+impl Tracer {
+    /// Create a new tracer.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// Tracer.new().trace_filename("mytrace.ptt").data_bufsize(1024).target_pid(666);
+    /// ```
+    pub fn new() -> Self {
+        Tracer {
+            trace_filename: CString::new("traceme.ptt").unwrap(),
+            target_pid: unsafe { getpid() },
+            tracer_ctx: None,
+            data_bufsize: 64,
+            aux_bufsize: 1024,
+        }
+    }
+
+    /// Set where to write the trace.
+    ///
+    /// # Arguments
+    ///
+    /// * `filename` - Filename to use.
+    pub fn trace_filename(mut self, filename: &str) -> Self {
+        self.trace_filename = CString::new(filename).expect("bad trace filename");
+        self
+    }
+
+    /// Select which PID to trace.
+    ///
+    /// By default, the current PID is traced.
+    ///
+    /// # Arguments
+    ///
+    /// * `pid` - PID to trace.
+    pub fn target_pid(mut self, pid: pid_t) -> Self {
+        self.target_pid = pid;
+        self
+    }
+
+    /// Set the PT data buffer size.
+    ///
+    /// # Arguments
+    ///
+    /// * `size` - The size to use.
+    pub fn data_bufsize(mut self, size: usize) -> Self {
+        self.data_bufsize = size;
+        self
+    }
+
+    /// Set the PT aux buffer size.
+    ///
+    /// # Arguments
+    ///
+    /// * `size` - The size to use.
+    pub fn aux_bufsize(mut self, size: usize) -> Self {
+        self.aux_bufsize = size;
+        self
+    }
+
+    // Make the map filename by setting/adding a ".map" extension to the trace filename
+    fn make_map_filename(&self, trace_filename: &CString) -> CString {
+        let map_string = trace_filename.clone().into_string().unwrap();
+        let mut map_path = PathBuf::from(map_string);
+        if !map_path.set_extension("map") {
+            panic!("failed to construct map filename");
+        }
+        let map_string = map_path.to_str().and_then(|ms| CString::new(ms).ok());
+        map_string.expect("failed to construct map filename")
+    }
+
+    /// Records execution of the selected PID into the chosen output file.
+    ///
+    /// Tracing continues until [stop_tracing](struct.Tracer.html#method.stop_tracing) is called.
```

Comment:
You can just say `until 'stop_tracing' is called` (where the `'` characters are reverse backticks) -- you don't need to do the full(ish) URL dance.

Reply:
That's not right.

Now:
![1](https://user-images.githubusercontent.com/604955/33261543-fc8a8dc4-d35a-11e7-8960-53c8174f897c.png)

With:
```diff
-    /// Tracing continues until [stop_tracing](struct.Tracer.html#method.stop_tracing) is called.
+    /// Tracing continues until `stop_tracing` is called.
```

![2](https://user-images.githubusercontent.com/604955/33261575-1123a824-d35b-11e7-8142-c5672c48f7c0.png)

I'm certain you need the URL to have the clickable link, we spoke in #rust about this on Friday afternoon.

Reply:
Oh... I stand corrected!